src/Components/CreateImageWizard/formComponents/Repositories.js

```diff
@@ -269,14 +278,23 @@ const Repositories = (props) => {
   };
 
   const filteredRepositoryURLs = useMemo(() => {
```

Comment:
We could remove a tiny bit of duplicity here by taking the `repoUrls` as a base and map it based on the clicked toggle. Something like this:

```javascript
  const filteredRepositoryURLs = useMemo(() => {
    const repoUrls = Object.values(repositories).filter((repo) =>
      repo.name.toLowerCase().includes(filterValue.toLowerCase())
    );
    if (toggleSelected === 'toggle-group-all') {
      return repoUrls.map((repo) => repo.url);
    } else if (toggleSelected === 'toggle-group-selected') {
      return repoUrls
        .filter((repo) => isRepoSelected(repo.url))
        .map((repo) => repo.url);
    }
  }, [filterValue, repositories, toggleSelected]);
```

What do you think?

Reply:
great idea